### ISSUE TITLE
[bug] - Grid height and width are not set

### ISSUE THREAD

**Issue by jmzagorski** [ghiscoding/aurelia-slickgrid#35]
**I'm submitting a bug report**

* **Framework Version:**
1.10.0

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Browser:**
Tested on Chrome/Firefox, but should be a bug on all

* **Language:**
Typescript

**Current behavior:**
The grid height and width are set using the `bindingContext` parameter on the `bind` method. However, the `bindingContext` will be my view model, which does not have a `gridHeight` or `gridWidth` properties because  I am setting the height and width directly in the view like so:

```html
<template>
  <aurelia-slickgrid 
    grid-id="elements-setup" 
    column-definitions.bind="elementColumns" 
    grid-options.bind="gridOptions" 
    dataset.bind="elements"
    grid-height="400" grid-width="800">
  </aurelia-slickgrid>
</template>
```

when inspecting `binding.gridHeight` and `binding.gridWidth`  [here](https://github.com/ghiscoding/aurelia-slickgrid/blob/master/aurelia-slickgrid/src/aurelia-slickgrid/aurelia-slickgrid.ts#L189), they are both undefined. They are undefined in Example1 too, so I am not sure how the height and width are being set to 800 and 400.

* **What is the expected behavior?**
Should use `this.gridHeight` and `this.gridWidth` since those will always have the values. Also you can probabaly remove `gridWidthString` and `gridHeightString` properties since they are not in use

**Comment by jmzagorski** [comment]
Actually you are resizing the grid [here](https://github.com/ghiscoding/aurelia-slickgrid/blob/master/aurelia-slickgrid/src/aurelia-slickgrid/aurelia-slickgrid.ts#L386) so maybe I am confused why the `style` property is being used?

**Comment by ghiscoding** [comment]
I can't fully recall, but doing the `${binding.gridHeight}px` might be because I tried to be in sync with my Angular repo. I haven't used much of the `binding` object from the `bind()` function yet. So yeah I could be totally off there. 

Your last comment with this [line](https://github.com/ghiscoding/aurelia-slickgrid/blob/master/aurelia-slickgrid/src/aurelia-slickgrid/aurelia-slickgrid.ts#L386) might have been added because the `${binding.gridHeight}px` wasn't working. 

I would suggest to fix it like you said with `this.gridHeight` and remove the resizer call (this [line](https://github.com/ghiscoding/aurelia-slickgrid/blob/master/aurelia-slickgrid/src/aurelia-slickgrid/aurelia-slickgrid.ts#L386)). I rarely use fixed size and prefer autoResize as much as possible.

**Comment by jmzagorski** [comment]
I was using fixed size because I was trying to put two grids in one view model (one on top of the the other), but only one grid would render. Have you ever tried that?

**Comment by ghiscoding** [comment]
Last time I tried in Angular it didn't work well (haven't tried in Aurelia). All the Services were using the same DI across the App, and I remember that I could see the 2 grids but certain actions done in 1st grid where in fact happening in the 2nd grid. I found the way to have separate Service instances in Angular, I assume there would be something similar in Aurelia?

**Comment by ghiscoding** [comment]
closed by #40

**Comment by ghiscoding** [comment]
@jmzagorski 
I tried to update the Github Demo and the first grid has this fixed height/width and it works in Dev mode but doesn't work after a Github Demo Buid (dist). Have you tried doing a dist build with a fixed size grid?

**Comment by jmzagorski** [comment]
I have not. I looked at it with `npm run start:dev`

**Comment by ghiscoding** [comment]
@jmzagorski 
Seems to be a problem with Edge browser only, not sure if it was working or not before the release. 

What I found so far is that in Edge, the `slick-viewport` has `1px` height. All other grids, which have auto-resize, are working as expected. I will have to do some more testing... I thought the IE issues were things of the past, but now Edge is back at it :|

**Comment by ghiscoding** [comment]
This was a bitch to fix, seems like `css.bind` is not fully working or at least not ideal in this lib. For some unknown reason, it seems like the `css.bind` was providing the sizes too late for SlickGrid to catch them, SlickGrid does need to know them, else it will hide the grid (they hide it through setting the `slick-viewport` to `1px` which is what I saw in Edge). SlickGrid reads these sizes and use them for calculating the viewport, headerrow, .... and then it uses jQuery to set each component with their available space. 

So all in all, I manage to resolve this by using `css` (instead of `css.bind`) with interpolation. I don't know why it works better but with all my tests (like 2 hours on this thing), it makes a difference. 

Also note, it wasn't only Edge, Firefox was also not working while Chrome was the only working browser (which I always develop with). 

Final note, when using autoResize, user won't provide a `gridHeight` and `gridWidth` which works better for the resize. If user does provide these sizes **and** uses `autoResize`, we can visually see a flickering caused by the resize of previous default size to the new size... so please don't provide `gridHeight` and `gridWidth` if you are not intending to use them.

I pushed a new patch release `1.11.1`, updated the Github Demo page and tested in all Edge/Firefox/Chrome.

**Comment by jmzagorski** [comment]
oh wow, wish I could have helped you out more on that one. I did a quick search this morning for css with Aurelia because I can never remember the correct way to do this and it is probably better to use string interpolation anyway because their docs discuss that http://aurelia.io/docs/binding/class-and-style#style

**Comment by ghiscoding** [comment]
Yeah string interpolation seems to be the most reliable because objects are not watched on property change. Anyhow in my case, it seems that using `css.bind` in the `bind( )` arrives too late in the process for SlickGrid to get the new sizes, even though technically `bind( )` is before `attached( )`. Also weird that `css` seems to happen before `css.bind` in lifecycle.

Anyway, my new fix works because I use `gridHeight` and `gridWidth` which are set from the start.

There's more info about the implementation of style binding in this [Github issue](https://github.com/aurelia/binding/issues/38#issuecomment-77398395).
